Source/WebGPU/WGSL/ConstantValue.h

```diff
@@ -36,22 +36,38 @@ namespace WGSL {
 #if HAVE(FP16_HALF_SUPPORT)
 using half = __fp16;
 #else
+// Wrap _Float16, which is the 16bit float type in C++23
 struct half {
+    using f16 = _Float16;
```

Comment:
will using `_Float16` instead of `__fp16` give different results? Since `__fp16` gets promoted to float for operations, but I don't know if we rely on this.

@tadeuzagallo do you recall?

Reply:
So it appears `_Float16` is promoted to `float` similar to `__fp16`. So I think it will be fine

Reply:
I remember we tried multiple different types for f16, but can't remember the details. I think the issue might have been when using `__half` instead. I can check locally that the wgslc tests still pass with the patch.

Reply:
I had to make some changes to get it compiling, but after that I ran all wgslc tests and it looks like it's fine.